src/main/java/org/jabref/model/database/BibDatabase.java

```diff
@@ -236,15 +235,15 @@ public synchronized void insertEntries(List<BibEntry> newEntries, EntriesEventSo
             internalIDs.add(id);
             entry.registerListener(this);
 
-            eventBus.post(new EntryAddedEvent(entry, eventSource));
-
             if (firstEntry == null) {
                 firstEntry = entry;
             }
         }
         entries.addAll(newEntries);
```

Comment:
I think, this should be wrapped inside `if (!newEntries.isEmpty())` and then inside no check for `firstEntry == null`. Then `firstEntry` is not required any more.

Reply:
For some reason I have a different version than you, where that is already deleted. Are you sure we are looking at the same version? Either way, from my understanding we need `firstEntry` because the way my version has it, `firstEntry` is passed into `EntriesAddedEvent` and expected by a listener in `org.jabref.gui.maintable.MainTable`.

Reply:
I try to comment on the source directly:

![grafik](https://user-images.githubusercontent.com/1366654/70868338-2f6a6c00-1f80-11ea-8661-7be9376b7099.png)

Maybe, instead of `firstEntry` something like `newEntries.get(0)` can be passed? - The `firstEntry` thing seems to be a hack